modules/dynamicfusion/CMakeLists.txt

```diff
@@ -1,82 +1,65 @@
-cmake_minimum_required(VERSION 3.5.0)
-
-# ---[ Configurations types
-set(CMAKE_CONFIGURATION_TYPES "Debug;Release" CACHE STRING "Possible configurations" FORCE)
+set(the_description "Non-rigid 3D reconstruction")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -DFORCE_INLINERS -D_MWAITXINTRIN_H_INCLUDED")
-if (DEFINED CMAKE_BUILD_TYPE)
-  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS ${CMAKE_CONFIGURATION_TYPES})
-endif()
-if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/Doxyfile)
-  find_package(Doxygen)
-  if(DOXYGEN_FOUND)
-
-    set(DOXYGEN_INPUT ${SOURCE_FILES})
-    set(DOXYGEN_OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile)
-
-    add_custom_command(
-            OUTPUT ${DOXYGEN_OUTPUT}
-            COMMAND ${CMAKE_COMMAND} -E echo_append "Building API Documentation..."
-            COMMAND ${DOXYGEN_EXECUTABLE} ${DOXYGEN_INPUT}
-            COMMAND ${CMAKE_COMMAND} -E echo "Done."
-            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-            DEPENDS ${DOXYGEN_INPUT}
-    )
-
-    add_custom_target(apidoc ALL DEPENDS ${DOXYGEN_OUTPUT})
-
-    add_custom_target(apidoc_forced
-            COMMAND ${CMAKE_COMMAND} -E echo_append "Building API Documentation..."
-            COMMAND ${DOXYGEN_EXECUTABLE} ${DOXYGEN_INPUT}
-            COMMAND ${CMAKE_COMMAND} -E echo "Done."
-            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-            )
-
-  endif(DOXYGEN_FOUND)
-endif()
 
-# ---[ Solution name
-project(kfusion C CXX)
-
-# ---[ utility
+#
+## ---[ Solution name
+#
+## ---[ utility
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/Modules/")
 include(cmake/Utils.cmake)
-include(cmake/Targets.cmake)
-
-# ---[ find dependencies
-find_package(OpenCV REQUIRED COMPONENTS core viz highgui calib3d)
+#include(cmake/Targets.cmake)
+#
+## ---[ find dependencies
 find_package(CUDA REQUIRED)
```

Comment:
You should not use `REQUIRED` option, instead just check whether the dependency has been found and disable your module if it can not be built without it:
```
find_package(Ceres QUIET)
if (NOT Ceres_FOUND)
    message(STATUS "${module} has been disabled because Ceres library has not been found")
    ocv_module_disable(${module})
endif()

Reply:
Try to use existing `HAVE_CUDA` variable, check the documentation at the beginning of https://github.com/opencv/opencv/blob/master/cmake/FindCUDA.cmake